src/app/helpers/helperMethods.ts

```diff
@@ -1,37 +1,123 @@
+                return reject(err);
+            }
+        });
     }
-};
 
-export function doesProjectFolderExists (projectFolder: string)
-{      
-  if (fs.existsSync(projectFolder))
-    {
-      if (fs.readdirSync(projectFolder).length > 0)
-      {          
-        return true;
-      }
+    function _updateMoveProjectFiles(projectFolder: string, host: string, typescript: boolean) {
+        return new Promise(async (resolve, reject) => {
+            // copy host-specific manifest over manifest.xml
+            fs.readFile(path.resolve(`${projectFolder}/manifest.${host}.xml`), 'utf8', (err, contents) => {
+                if (err) return reject(err);
+                fs.writeFile(path.resolve(`${projectFolder}/manifest.xml`), contents, (err) => {
+                    if (err) return reject(err);
+                });
+            });
+
+            // copy host-specific taskpane.ts over src/taskpane/taskpane.ts[js]
+            fs.readFile(path.resolve(`${projectFolder}/src/taskpane/${host}.${typescript ? 'ts' : 'js'}`), 'utf8', (err, contents) => {
+                if (err) return reject(err);
+                fs.writeFile(path.resolve(`${projectFolder}/src/taskpane/taskpane.${typescript ? 'ts' : 'js'}`), contents, (err) => {
+                    if (err) return reject(err);
+                });
+            });
+
+            // delete all host specific files
+            hosts.forEach(function (host) {
+                fs.unlink(path.resolve(`${projectFolder}/manifest.${host}.xml`), (err) => {
+                    if (err) return reject(err);
+                });
+                fs.unlink(path.resolve(`${projectFolder}/src/taskpane/${host}.${typescript ? 'ts' : 'js'}`), (err) => {
+                    if (err) return reject(err);
+                });
+            });
+            return resolve();
+        });
+    }
+
+    function _modifyPackageJsonFile(projectFolder:string, host: string) {
+        return new Promise(async (resolve, reject) => {
+            // update package.json to reflect selected host
+            const packageJson = path.resolve(`${projectFolder}/package.json`);
+            fs.readFile(packageJson, 'utf8', (err, data) => {
+                if (err) return reject(err);
+                let content = JSON.parse(data);
+
+                // update 'config' section in package.json to use selected host
+                content.config["app-to-debug"] = host;
+
+                // remove scripts from package.json that are unrelated to selected host,
+                // and update sideload and unload scripts to use selected host.
+                Object.keys(content.scripts).forEach(function (key) {
+                    if (key.includes("sideload:") || key.includes("unload:")) {
+                        delete content.scripts[key];
+                    }
+
+                    if (key == "sideload") {
+                        content.scripts[key] = `office-toolbox sideload -m manifest.xml -a ${host}`
```

Comment:
You should not write out the scripts value here. It should come from the project. You just need to copy the host-specific value to the general value:
```
scripts.[key] = scripts.[`${key}:${host}`];
```

This works for both `sideload` and `unload` scripts so just need switch around it:
```
switch (key) {
   case "sideload":
   case "unload":
     scripts.[key] = scripts.[`${key}:${host}`];
     break;
}
```